### DISCUSSION TITLE
Set file permissions with deb package

### DISCUSSION THREAD

**Discussion by keastrid** [fvarrui/JavaPackager#420] [Ideas]
It would be useful to be able to set file permissions with the deb installation package, so that the app can have control over its own files (eg. logging, self updating, etc). Apparently, this is done via a "rules" file that gets added to the package.

**Comment by fvarrui** [comment]
Not sure if it's a good idea. To upgrade your app you need "root" privileges, and [log files should be stored in `/var/log` or in the user profile](https://stackoverflow.com/a/1510352).

**Comment by fvarrui** [comment]
And [here](https://stackoverflow.com/a/4002730) is explained how a self-upgreadable app should work ... maybe, in your app's entry point or after push a button, your app should check if there's a new version, and if so, it will download and install the new version with elevated privileges. Or just create a portable app, installed into the user's profile.

**Comment by keastrid** [comment]
We have updating already working (its done on user action, not automatically). It might be best (for us) to leave it as portable only.
